Polyhedron/demo/Polyhedron/Travel_isolated_components.h

```diff
@@ -5,12 +5,58 @@
 #include <vector>
 #include "One_ring_iterators.h"
 #include <CGAL/Default.h>
+
+
+template<typename Mesh>
+struct Id_getter{
+  typedef typename boost::graph_traits<Mesh>::vertex_descriptor mesh_vd;
+  typedef typename boost::graph_traits<Mesh>::face_descriptor mesh_fd;
+  typedef typename boost::graph_traits<Mesh>::halfedge_descriptor mesh_hd;
+  typedef typename boost::graph_traits<Mesh>::edge_descriptor mesh_ed;
+  const Mesh& mesh;
+  typename boost::property_map<Mesh, boost::vertex_index_t >::type vmap;
+  typename boost::property_map<Mesh, boost::face_index_t >::type fmap;
+  typename boost::property_map<Mesh, boost::halfedge_index_t >::type hmap;
+  typename boost::property_map<Mesh, boost::edge_index_t >::type emap;
+
+  Id_getter(const Mesh& mesh)
+    :mesh(mesh)
+  {
+    vmap = get(boost::vertex_index, mesh);
+    fmap = get(boost::face_index, mesh);
+    hmap = get(boost::halfedge_index, mesh);
+    emap = get(boost::edge_index, mesh);
+  }
+
+  std::size_t id(mesh_vd vd)
+  {
+    return get(vmap, vd);
+  }
+  std::size_t id(mesh_fd fd)
+  {
+    return get(fmap, fd);
+  }
+  std::size_t id(mesh_hd hd)
+  {
+    return get(hmap, hd);
+  }
+  std::size_t id(mesh_ed ed)
+  {
+    return get(emap, ed);
+  }
```

Comment:
Remark for @afabri and @sloriot, not really related to this PR: that sort of API will fail as soon as we have a model of `FaceGraph` that use simple integers for all types of descriptor. An API assuming that, for example, `vertex_descriptor` and `face_descriptor` are different, cannot be generic.